Detectors/TPC/reconstruction/include/TPCReconstruction/ClustererTask.h

```diff
@@ -18,68 +18,93 @@
       switch (type) {
         case ClustererType::HW:   return mHwClustererEnable;
         case ClustererType::Box:  return mBoxClustererEnable;
       };
     };
 
-    Clusterer* getClusterer(ClustererType type) { 
+    /// Returns pointer to requested Clusterer type
+    /// \param type - Clusterer type, HW or Box
+    /// \return  Pointer to Clusterer, nullptr if Clusterer was not enabled during Init()
+    Clusterer* getClusterer(ClustererType type) {
       switch (type) {
-        case ClustererType::HW:   return mHwClusterer;
-        case ClustererType::Box:  return mBoxClusterer;
+        case ClustererType::HW:   return mHwClusterer.get();
+        case ClustererType::Box:  return mBoxClusterer.get();
       };
     };
+
+    /// Returns pointer to Box Clusterer
+    /// \return  Pointer to Clusterer, nullptr if Clusterer was not enabled during Init()
+    BoxClusterer* getBoxClusterer()   const { return mBoxClusterer.get(); };
     
-    BoxClusterer* getBoxClusterer()   const { return mBoxClusterer; };
-    HwClusterer* getHwClusterer()     const { return mHwClusterer; };
-    //             Clusterer *GetClusterer() const { return fClusterer; }
-    
+    /// Returns pointer to Box Clusterer
+    /// \return  Pointer to Clusterer, nullptr if Clusterer was not enabled during Init()
+    HwClusterer* getHwClusterer()     const { return mHwClusterer.get(); };
+
   /// Switch for triggered / continuous readout
   /// \param isContinuous - false for triggered readout, true for continuous readout
   void setContinuousReadout(bool isContinuous);
 
   private:
-    bool          mBoxClustererEnable;
-    bool          mHwClustererEnable;
-    bool          mIsContinuousReadout; ///< Switch for continuous readout
+    bool mBoxClustererEnable;   ///< Switch to enable Box Clusterfinder
+    bool mHwClustererEnable;    ///< Switch to enable Hw Clusterfinder
+    bool mIsContinuousReadout;  ///< Switch for continuous readout
+    int mEventCount;            ///< Event counter
+
+    std::unique_ptr<BoxClusterer> mBoxClusterer;    ///< Box Clusterfinder instance
+    std::unique_ptr<HwClusterer> mHwClusterer;      ///< Hw Clusterfinder instance
+
+    // Digit arrays
+    std::vector<o2::TPC::Digit> const *mDigitsArray;    ///< Array of TPC digits
+    MCLabelContainer const *mDigitMCTruthArray;         ///< Array for MCTruth information associated to digits in mDigitsArrray
+
+    // Cluster arrays
+    std::vector<o2::TPC::Cluster> *mClustersArray;              ///< Array of clusters found by Box Clusterfinder
+    std::vector<o2::TPC::Cluster> *mHwClustersArray;            ///< Array of clusters found by Hw Clusterfinder
+    std::unique_ptr<MCLabelContainer> mClustersMCTruthArray;      ///< Array for MCTruth information associated to cluster in mClustersArrays
+    std::unique_ptr<MCLabelContainer> mHwClustersMCTruthArray;    ///< Array for MCTruth information associated to cluster in mHwClustersArrays
```

Comment:
just out of curiosity, is the clusterization running on simulated and HW clusters in the same process so that it makes sense to carry the two types of arrays? Or is it more a kind of a specialization since the task is running on either the one or the other?

Reply:
There is still this BoxClusterer, which could in principle also be used. Although I think the BoxClusterer doesn't work anymore.